compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1268,19 +1261,16 @@ OMR::ResolvedMethodSymbol::genIL(TR_FrontEnd * fe, TR::Compilation * comp, TR::S
             if (doOSR)
                {
                TR_ASSERT(comp->getOSRCompilationData(), "OSR compilation data is NULL\n");
-               if (comp->canAffordOSRControlFlow())
+               self()->genAndAttachOSRCodeBlocks(comp->getCurrentInlinedSiteIndex());
```

Comment:
The addition of [osrInfrastructureRemoved](https://github.com/eclipse/omr/commit/955f4104b8f026fdb21c8047c746586e22399061) added a call to supportsInduceOSR() to doOSR, which will check both canAffordOSRControlFlow and osrInfrastructureRemoved. The calls between the evaluation of doOSR and the check create an ILGen optimizer and detecting cycles in the CFG. Neither of these will modify the flags as canAffordOSRControlFlow is never modified and osrInfrastructureRemoved is only changed if OSRExceptionEdgeRemoval is performed. Therefore, the check was redundant.